test/e2e/metrics_kubeletstats_test.go

```diff
@@ -0,0 +1,190 @@
+)
+
+var _ = Describe("Metrics Kubeletstats", Label("metrics"), func() {
+	Context("When a metricpipeline exists", Ordered, func() {
+		var (
+			pipelines          *kyma.PipelineList
+			urls               *mocks.URLProvider
+			mockDeploymentName = "metric-agent-receiver"
+			mockNs             = "metric-agent-mocks"
+			metricGatewayName  = types.NamespacedName{Name: metricAgentGatewayBaseName, Namespace: kymaSystemNamespaceName}
+		)
+
+		BeforeAll(func() {
+			k8sObjects, urlProvider, pipelinesProvider := makeMetricsAgentTestK8sObjects(mockNs, mockDeploymentName)
+			pipelines = pipelinesProvider
+			urls = urlProvider
+
+			DeferCleanup(func() {
+				Expect(kitk8s.DeleteObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+			})
+			Expect(kitk8s.CreateObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+		})
+
+		It("Should have a running metric gateway deployment", Label(operationalTest), func() {
+			Eventually(func(g Gomega) {
+				ready, err := verifiers.IsDeploymentReady(ctx, k8sClient, metricGatewayName)
+				g.Expect(err).ShouldNot(HaveOccurred())
+				g.Expect(ready).To(BeTrue())
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("Should have a metrics backend running", Label(operationalTest), func() {
+			Eventually(func(g Gomega) {
+				key := types.NamespacedName{Name: mockDeploymentName, Namespace: mockNs}
+				ready, err := verifiers.IsDeploymentReady(ctx, k8sClient, key)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(ready).To(BeTrue())
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("Should have a running agent daemonset", Label(operationalTest), func() {
+			Eventually(func(g Gomega) bool {
+				var daemonSet appsv1.DaemonSet
+				key := types.NamespacedName{Name: metricAgentBaseName, Namespace: kymaSystemNamespaceName}
+				g.Expect(k8sClient.Get(ctx, key, &daemonSet)).To(Succeed())
+
+				listOptions := client.ListOptions{
+					LabelSelector: labels.SelectorFromSet(daemonSet.Spec.Selector.MatchLabels),
+					Namespace:     kymaSystemNamespaceName,
+				}
+				var pods corev1.PodList
+				g.Expect(k8sClient.List(ctx, &pods, &listOptions)).To(Succeed())
+				for _, pod := range pods.Items {
+					for _, containerStatus := range pod.Status.ContainerStatuses {
+						g.Expect(containerStatus.State.Running).NotTo(BeNil())
+					}
+				}
+
+				return true
```

Comment:
We have a readiness check for deployments (`verifiers.IsDeploymentReady`). Let's do the same for a DaemonSet